ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -13,24 +65,236 @@ class RagasFaithfulnessEvaluator:
     METRIC_TYPE = "score"
 
     def __init__(self, llmobs_service):
+        """
+        Initialize an evaluator that uses the ragas library to generate a faithfulness score on finished LLM spans.
+
+        Faithfulness measures the factual consistency of an LLM's output against a given context.
+        There are two LLM calls required to generate a faithfulness score - one to generate a set of statements from
+        the answer, and another to measure the faithfulness of those statements against the context using natural
+        language entailment.
+
+        For more information, see https://docs.ragas.io/en/latest/concepts/metrics/faithfulness/
+
+        The `ragas.metrics.faithfulness` instance is used for faithfulness scores. If there is not llm attribute set
+        on this instance, it will be set to the default `llm_factory()` which uses openai.
+
+        :param llmobs_service: An instance of the LLM Observability service used for tracing the evaluation and
+                                submitting evaluation metrics.
+        """
+        self.enabled = True
+        if not RAGAS_DEPENDENCIES_PRESENT:
+            self.enabled = False
+            return
+
         self.llmobs_service = llmobs_service
+        self.ragas_faithfulness_instance = _get_faithfulness_instance()
+
+        if not self.ragas_faithfulness_instance.llm:
+            self.ragas_faithfulness_instance.llm = llm_factory()
+
+        self.llm_output_parser_for_generated_statements = RagasoutputParser(pydantic_object=StatementsAnswers)
+
+        self.llm_output_parser_for_faithfulness_score = RagasoutputParser(pydantic_object=StatementFaithfulnessAnswers)
 
-    def run_and_submit_evaluation(self, span: dict) -> None:
-        if not span:
+        self.split_answer_into_sentences = get_segmenter(
+            language=self.ragas_faithfulness_instance.nli_statements_message.language, clean=False
+        )
+
+    def run_and_submit_evaluation(self, span_event: dict):
+        if not span_event:
             return
-        score_result = self.evaluate(span)
+        score_result = self.evaluate(span_event)
         if score_result:
             self.llmobs_service.submit_evaluation(
-                span_context=span,
+                span_context={"trace_id": span_event.get("trace_id"), "span_id": span_event.get("span_id")},
                 label=RagasFaithfulnessEvaluator.LABEL,
                 metric_type=RagasFaithfulnessEvaluator.METRIC_TYPE,
                 value=score_result,
-                timestamp_ms=math.floor(time.time() * 1000),
```

Comment:
Why did we get rid of timestamping?

Reply:
There's the same logic inside `submit_evaluation` so i thought I'd just remove this line

`        timestamp_ms = timestamp_ms if timestamp_ms else int(time.time() * 1000)
`